Source/IGListDiff.mm

```diff
@@ -228,29 +237,64 @@ static id IGListDiffing(BOOL returnIndexPaths,
             addIndexToCollection(mInserts, toSection, i);
             runningOffset++;
         } else {
+            oldInsertOffsets[i - runningOffset + deleteOffsets[oldIndex]] = runningOffset;
```

Comment:
Can you explain the theory behind this? Spent some time this morning wrapping my head around these changes and this is the only thing leaving me puzzled.